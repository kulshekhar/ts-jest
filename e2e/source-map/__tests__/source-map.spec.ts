import { sync as spawnSync } from 'execa'
import stripAnsi from 'strip-ansi'

describe('source-map', () => {
  it('should display failed test with stacktrace', () => {
    // Use `--json` flag to ignore differences between OS when creating snapshots
    const { stdout } = spawnSync('jest', ['-c=e2e/source-map/src/jest-fake.config.js', '--json'], {
      cwd: process.cwd(),
      reject: false,
    })

    const message = JSON.parse(stdout || '').testResults[0].message

    expect(stripAnsi(message)).toMatchSnapshot()
  })
})
