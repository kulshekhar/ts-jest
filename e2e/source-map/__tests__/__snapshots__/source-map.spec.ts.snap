// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`source-map should display failed test with stacktrace 1`] = `
"  ● should fail

    expect(received).toBe(expected) // Object.is equality

    Expected: false
    Received: true

      1 | it('should fail', () => {
    > 2 |   expect(true).toBe(false)
        |                ^
      3 | })
      4 |

      at Object.toBe (error.test-error.js:2:16)
"
`;
